src/openclsim/critical_path/simulation_graph.py

```diff
@@ -37,12 +41,53 @@ class SimulationGraph:
         should appear in the list as tuple ``(1, 2)``.
     """
 
+    # name of added columns to the recorded_activities_df
+    __COLNAME_CRITICAL = "is_critical"
+
+    # mapping from column name to attribute
+    __SUPERLOG_COLUMNS = {
+        "Activity": "activity",
+        "SimulationObject": "source_object",
+        "start_time": "start_time",
+        "state": "state",
+        "duration": "duration",
+        "end_time": "end_time",
+        "cp_activity_id": "cp_activity_id",
+    }
+
+    # fixed edge types
+    __EDGE_TYPES = {0: "activity", 1: "dependency"}
+
+    # criticality
+    __CRITICAL = {0: False, 1: True}
+
+    # prefixes to use for node start and end
+    __NODE_START_PREFIX = "start"
+    __NODE_END_PREFIX = "end"
+
     def __init__(self, recorded_activity_df, dependency_list):
-        """Init"""
+        """Initiate the object."""
+        self.critical_edges_list = None
         self.recorded_activity_df = self.__check_rec(recorded_activity_df)
         self.dependency_list = self.__check_dep(dependency_list)
+
+        # construct the graph based on the log and dependencies
         self.simulation_graph = self.__construct_graph()
-        self.__find_critical_edges()
+
+        self.n_activities = len(
+            [
+                e
+                for e in self.simulation_graph.edges
+                if self.simulation_graph.edges[e]["edge_type"] == "activity"
+            ]
+        )
+
+        # get the duration of longest path
+        self.max_duration = nx.dag_longest_path_length(
+            self.simulation_graph, weight="duration"
+        )
+
+        # self.__find_critical_edges()
 
     def __check_rec(self, recorded_activity_df):
```

Comment:
should this be a full name? e.g. check_recorded_activity_df

Reply:
best do so!